### PR TITLE
Request to add DEFI project AstroSwap

### PR DESCRIPTION
**Pull request by Astro-swap**
Name (to be shown on DefiLlama):
AstroSwap

Twitter [Link:](https://x.com/astroswapreal)
https://x.com/astroswapreal

List of audit links if any:
no

Website Link:
https://astroswap.xyz/

Logo (High resolution, will be shown with rounded borders):

Current TVL:
112629

Treasury Addresses (if the protocol has treasury)
no

Chain:
starknet

Coingecko ID (so your TVL can appear on Coingecko, leave empty if not listed): (https://api.coingecko.com/api/v3/coins/list)
no

Coinmarketcap ID (so your TVL can appear on Coinmarketcap, leave empty if not listed): (https://api.coinmarketcap.com/data-api/v3/map/all?listing_status=active,inactive,untracked&start=1&limit=10000)
no

Short Description (to be shown on DefiLlama):
AstroSwap——AMM,Liquidity Farms,and Reward pools on Starknet. 

Token address and ticker if any:
no

Category (full list at https://defillama.com/categories) *Please choose only one:
Dexes

Oracle used (Chainlink/Band/API3/TWAP or any other that you are using):
No

forkedFrom (Does your project originate from another project):

methodology (what is being counted as tvl, how is tvl being calculated):
tvl

### CODE REVIEW

File: projects/astosswap/api.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+const ADDRESSES = require('../helper/coreAssets.json')
+
+const { call, multiCall, parseAddress } = require('../helper/chain/starknet')
+const abi = require('./abi')
+const { transformDexBalances } = require('../helper/portedTokens')
+const { getParamCalls } = require('../helper/utils')
+
+async function tvl() {
+  const factory = '0x07df3bce30857e8f9c08bcd9d9668df34166e94dd968db6e2920b870c4410e34'
```
Comment (g1nt0ki) [inline projects/astosswap/api.js:9]: this is stakex factory address

Comment (Astro-swap) [reply projects/astosswap/api.js:9]: Yes 
we are StarkEx AMM team 
our previous project website and twitter is suspended
we just deploy product on this new domain
pls check

Comment (g1nt0ki) [reply projects/astosswap/api.js:9]: I see that the discord is still active, can you post the new domain/rebranding info there from admin account as proof?

Comment (Astro-swap) [reply projects/astosswap/api.js:9]: Sure We post it within 5 minutes

Comment (Astro-swap) [reply projects/astosswap/api.js:9]: Just post the announcement
image
pls check there

**Comment by Define101** [comment]
There is a protocol called AstroSwap already
Would you prefer Astro Swap or AstroSwap AMM, or any other name?

**Comment by Astro-swap** [comment]
> There is a protocol called AstroSwap already
> Would you prefer Astro Swap or AstroSwap AMM, or any other name?

We choose AstroSwap AMM

File: projects/astosswap/api.js (hunk 1 above)
Comment (Astro-swap) [reply projects/astosswap/api.js:9]: We choose AstroSwap AMM

**Comment by Define101** [comment]
https://defillama.com/protocol/astroswap-amm

**Pull request closed without merging**